### DISCUSSION TITLE
Expand multiple relation

### DISCUSSION THREAD

**Discussion by renatocaliari** [pocketbase/pocketbase#2742] [Q&A] [answered]
Does `expand` work only for "single" relations, or also for "multiple" relations? 

I have the impression that when it is a "multiple" relationship, the field is returned as an array of strings and `expand` does not work for this case. If that is the case, do I need to read the data from the related table, filtering by the IDs returned in the array?

I may be totally mistaken, as I am still learning how to work with Pocketbase. :)

Thanks!

**Comment by ganigeorgiev** [comment]
Direct `relation` field expansion is supported with both single and multiple values.
Indirect expand (aka. when the relation field is not in the main collection) is supported as of now only for single relations.
If you haven't already, for more details and example you can check https://pocketbase.io/docs/expanding-relations/.

The syntax is the same for both single and multiple relations (use dot-notation for nested relations expansion)

```
await pb.collection("example").getList(1, 30, {
    expand: "multipleRelField.anotherSubRelField"
})
```

The result for multiple relations in the `expand` prop will be array of records.

If the above doesn't help, please provide a code sample that you've tried and more details about your collections structure (_you can export your collections as JSON from the Admin UI > Settings > Export collections_).

**Reply by renatocaliari** [reply]
Great to know, thanks. 
So, I'm probably doing something wrong. The issue is probably staring me in the face and I'm not seeing it.

Here are some images that might help to understand my context:

![Screenshot 2023-06-18 10 46 57](https://github.com/pocketbase/pocketbase/assets/1719651/3c07d6e8-4733-42b5-96ff-ec86c9e98e92)
![Screenshot 2023-06-18 10 46 43](https://github.com/pocketbase/pocketbase/assets/1719651/30aa73a0-a8e5-4a75-b26e-978c4c19d6fd)
--
### Roles:
![Screenshot 2023-06-18 10 48 06](https://github.com/pocketbase/pocketbase/assets/1719651/73da5ebe-5db0-4320-a010-5ac402cf645a)
![Screenshot 2023-06-18 10 52 22](https://github.com/pocketbase/pocketbase/assets/1719651/6cb8e5d1-d214-4419-966c-67ed183a5b21)

### Tags:
![Screenshot 2023-06-18 10 48 52](https://github.com/pocketbase/pocketbase/assets/1719651/ed8f3f27-d694-4d98-8c2d-c496cccd541b)

**Reply by ganigeorgiev** [reply, marked as answer]
Make sure that the client sending the request satisfies the `tags` collection View API rule.

**Reply by renatocaliari** [reply]
that's it! thanks.

**Reply by Oskar1504** [reply]
I faced this issue so oft now. Always forget to check the API rules.

**Comment by stefandanzl** [comment]
I thought I had a similar problem, but I only saw [Object] in the expanded properties when I did `console.log(record)`
which you can easily solve with `console.dir(record, { depth: null }); `. 
This makes sure it displays the full depth of your object.
